packages/graphback-runtime/src/resolvers/LayeredRuntimeResolverCreator.ts

```diff
@@ -69,13 +69,14 @@ export class LayeredRuntimeResolverCreator {
 
       if (resolverElement.crudOptions.findOne) {
         const findOneField = getFieldName(modelName, GraphbackOperationType.FIND_ONE);
+        const primaryKeyLabel = getPrimaryKey(resolverElement.graphqlType).name;
         //tslint:disable-next-line: no-any
         resolvers.Query[findOneField] = (parent: any, args: any, context: any) => {
           if (!this.services[modelName]) {
             throw new Error(`Missing service for ${modelName}`);
           }
 
-          return this.services[modelName].findOne(args, context)
+          return this.services[modelName].findOne({ [primaryKeyLabel]: args.id }, context)
```

Comment:
There needs to be an annotation provided to map the static `id` to the true primary key, for cases where the primary key is something different, for example `email`.

Reply:
Never mind I was misunderstanding it.

Reply:
My biggest issue with only performing ID mapping at resolver level is that this is not obvious from the schema. The user may not make a connection between the static id and their primary key.

```gql
type User {
  id: ID!
  """
  @db.primary
  """
  email: String
}
```

In this example, `findOne` will remap id to email, but from the user perspective they may expect it to fetch by their explicitly defined ID field.

We should additionally have an annotation to let the user add this mapping.

Reply:
Happy for you to take over with something different. 
I never actually liked get in the first place :D